packages/edit-site/src/components/sidebar-navigation-screen-pages/index.js

```diff
@@ -5,64 +5,157 @@ import {
+		templates?.find( ( template ) => template.slug === 'index' );
+
+	const pagesAndTemplates = pages?.concat( dynamicPageTemplates, [
+		homeTemplate,
+	] );
+
+	const { frontPage, postsPage } = useSelect( ( select ) => {
+		const { getEntityRecord } = select( coreStore );
+
+		const siteSettings = getEntityRecord( 'root', 'site' );
+		return {
+			frontPage: siteSettings?.page_on_front,
+			postsPage: siteSettings?.page_for_posts,
+		};
+	}, [] );
+
+	const isHomePageBlog = frontPage === postsPage;
+
+	const reorderedPages = pages && [ ...pages ];
+
+	if ( ! isHomePageBlog && reorderedPages?.length ) {
+		const homePageIndex = reorderedPages.findIndex(
+			( item ) => item.id === frontPage
+		);
+		const homePage = reorderedPages.splice( homePageIndex, 1 );
+		reorderedPages?.splice( 0, 0, ...homePage );
+
+		const postsPageIndex = reorderedPages.findIndex(
+			( item ) => item.id === postsPage
+		);
+
+		const blogPage = reorderedPages.splice( postsPageIndex, 1 );
+
+		reorderedPages.splice( 1, 0, ...blogPage );
+	}
 
 	return (
 		<SidebarNavigationScreen
 			title={ __( 'Pages' ) }
 			description={ __( 'Browse and edit pages on your site.' ) }
 			content={
 				<>
-					{ isLoading && (
+					{ ( isLoadingPages || isLoadingTemplates ) && (
 						<ItemGroup>
 							<Item>{ __( 'Loading pages' ) }</Item>
 						</ItemGroup>
 					) }
-					{ ! isLoading && (
-						<>
-							<SidebarNavigationSubtitle>
-								{ __( 'Recent' ) }
-							</SidebarNavigationSubtitle>
-							<ItemGroup>
-								{ ! pages?.length && (
-									<Item>{ __( 'No page found' ) }</Item>
-								) }
-								{ pages?.map( ( page ) => (
+					{ ! ( isLoadingPages || isLoadingTemplates ) && (
+						<ItemGroup>
+							{ ! pagesAndTemplates?.length && (
+								<Item>{ __( 'No page found' ) }</Item>
+							) }
+							{ isHomePageBlog && homeTemplate && (
+								<PageItem
+									postType="wp_template"
+									postId={ homeTemplate.id }
+									key={ homeTemplate.id }
+									icon={ home }
+									withChevron
+								>
+									<Truncate numberOfLines={ 1 }>
+										{ decodeEntities(
+											homeTemplate.title?.rendered
+										) ?? __( '(no title)' ) }
+									</Truncate>
+								</PageItem>
+							) }
+							{ reorderedPages?.map( ( item ) => {
+								let itemIcon;
+								switch ( item.id ) {
+									case frontPage:
+										itemIcon = home;
+										break;
+									case postsPage:
+										itemIcon = loop;
+										break;
+									default:
+										itemIcon = page;
+								}
+								return (
 									<PageItem
-										postId={ page.id }
-										key={ page.id }
+										postId={ item.id }
+										key={ item.id }
+										icon={ itemIcon }
 										withChevron
 									>
 										<Truncate numberOfLines={ 1 }>
 											{ decodeEntities(
-												page.title?.rendered
-											) ?? __( 'Untitled' ) }
+												item.title?.rendered
+											) ?? __( '(no title)' ) }
```

Comment:
https://github.com/WordPress/gutenberg/pull/50767 uses `'Untitled'` as fallback. 

I have no preference either way, but we should make them the same.

Reply:
"no title" is what the pages were already using, this PR only adds a couple more instances. Was changing it to "Untitled" a design requirement? I'm happy with either, but would prefer to address that as a follow-up.

Reply:
>  Was changing it to "Untitled" a design requirement? 

Not at all, just asking so that I can update the individual page view to match this copy 👍

Reply:
PR here: https://github.com/WordPress/gutenberg/pull/5